sourcePage.js

```diff
@@ -17,61 +22,177 @@ function getPage() {
         // Run script to get HTML after major has been retrieved from storage
         chrome.runtime.sendMessage({
             action: "getSource",
-            source: DOMtoString(selectedMajor)
+            source: getHtml(selectedMajor)
         });
     });
 }
 
 // Reads through HTML and gets class names, prints to popup with concentration
-function DOMtoString(selectedMajor) {
-    var html = '';
+function getHtml(selectedMajor) {
+    selectedMajor = selectedMajor.slice(1, -1); // Take quotes off string
+    var html = 'Major: ' + selectedMajor + '\n';
     var nodes = document.body.getElementsByClassName('classAbbreviation');
-    selectMajor = selectedMajor.substr(1, selectedMajor.length - 2); // Shave off quotation marks
-    html += ('My major is ' + selectedMajor + '\n');
 
     // Parse html of page
     var i = 0; 
     var l = nodes.length;
     for (i = 0; i < l; i++) {
-        var element = nodes[i].innerHTML;
-        var concentration; 
+        var className = nodes[i].innerHTML;
+        className = className.substr(0, className.length - 1); // Take off elipses from class name
+        var concentration; // Local variable to hold concentration
```

Comment:
use a slice for className or use a substr above when you remove quotes. Stay consistent.